File: 1346_check_if_n_and_its_double_exist/Solution.java
class Solution {
    public boolean checkIfExist(int[] arr) {
        HashSet<Integer> seen = new HashSet<Integer>();
        for(int i = 0; i<arr.length; i++){
            if(seen.contains(arr[i]*2) || (arr[i]%2 == 0 && seen.contains(arr[i]/2))){
                return true;
            }
            seen.add(arr[i]);
        }
        return false;
    }
}

File: 1346_check_if_n_and_its_double_exist/Solution.cpp
class Solution {
public:
    bool checkIfExist(vector<int>& arr) {
        unordered_set<int> seen;
        for(int i = 0; i<arr.size(); i++){
            if(seen.count(arr[i]*2)>0 || (arr[i]%2 == 0 && seen.count(arr[i]/2)>0)){
                return true;
            }
            seen.insert(arr[i]);
        }
        return false;
    }
};
